64_FFT/app_resources/common.hlsl

```diff
@@ -0,0 +1,23 @@
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+
+// Unfortunately not every piece of C++14 metaprogramming syntax is available in HLSL 202x
+// https://github.com/microsoft/DirectXShaderCompiler/issues/5751#issuecomment-1800847954
+typedef nbl::hlsl::float32_t2 input_t;
+typedef nbl::hlsl::float32_t output_t;
+
+NBL_CONSTEXPR_STATIC_INLINE uint32_t MaxPossibleElementCount = 1 << 20;
+
+struct PushConstantData
+{
+	uint64_t inputAddress;
+	uint64_t outputAddress;
```

Comment:
maybe use `bda::__ptr<>` or whatever we have

Reply:
ah yeah then you need a different definition for C++ and for HLSL, cause `bda::__ptr<>` not available in C++